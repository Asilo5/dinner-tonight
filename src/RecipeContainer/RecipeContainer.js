import React from 'react';
import Recipe from '../Recipe/Recipe';

const RecipeContainer = (props) => {
//   const recipes = props.map((recipe) => {
//       return <Recipe key={recipe.recipe_id} 
//                      id={recipe.recipe_id}
//                      title={recipe.title}
//                      />
//   })
  return (
    <section className='recipe-container'>
    {/* {recipes} */}
    <Recipe />
    </section>
  )
}

export default RecipeContainer;