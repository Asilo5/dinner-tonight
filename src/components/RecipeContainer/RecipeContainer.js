import React from 'react';
import './RecipeContainer.css';
import Recipe from '../Recipe/Recipe';
import PropTypes from 'prop-types';

const RecipeContainer = ({ recipes }) => {
  const recipe = recipes.map((recipe) => {
      return <Recipe key={recipe.recipe_id} 
                     id={recipe.recipe_id}
                     title={recipe.title}
                     publisher={recipe.publisher}
                     image={recipe.image_url}
                     recipeLink={recipe.source_url}
                     />
    })
  return ( 
    <section className='recipe-container'>
    {recipe}
    </section>
  )
}

Recipe.propTypes = {
  recipes: PropTypes.array
}

export default RecipeContainer;