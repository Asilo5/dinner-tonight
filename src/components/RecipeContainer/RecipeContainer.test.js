import React from 'react';
import { shallow } from 'enzyme';
import RecipeContainer from './RecipeContainer';

describe('RecipeContainer', () => {

    let wrapper;
    let mockRecipes = [
        {
            image_url: "http://static.food2fork.com/BananaPeanutButterChocolateChipCookies13e87.jpg",
            publisher: "Two Peas and Their Pod",
            publisher_url: "http://www.twopeasandtheirpod.com",
            recipe_id: "ed9409",
            source_url: "http://www.twopeasandtheirpod.com/gluten-free-vegan-banana-peanut-butter-chocolate-chip-cookies/",
            title: "Gluten-Free Vegan Banana Peanut Butter Chocolate Chip Cookies",
          }
    ];

    beforeEach(() => {
        wrapper = shallow(<RecipeContainer recipes={mockRecipes}/>)
    })

    it('should match snapshot with all information passing in correctly', () => {
        expect(wrapper).toMatchSnapshot();
    })
})