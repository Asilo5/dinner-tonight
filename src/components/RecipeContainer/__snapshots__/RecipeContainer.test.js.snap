// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipeContainer should match snapshot with all information passing in correctly 1`] = `
<section
  className="recipe-container"
>
  <Recipe
    id="ed9409"
    image="http://static.food2fork.com/BananaPeanutButterChocolateChipCookies13e87.jpg"
    key="ed9409"
    publisher="Two Peas and Their Pod"
    recipeLink="http://www.twopeasandtheirpod.com/gluten-free-vegan-banana-peanut-butter-chocolate-chip-cookies/"
    title="Gluten-Free Vegan Banana Peanut Butter Chocolate Chip Cookies"
  />
</section>
`;
